aiida/work/job_processes.py

```diff
@@ -507,6 +507,7 @@ def _setup_db_record(self):
         from aiida.common.links import LinkType
 
         self.calc._set_process_type(self._calc_class)
+        self.calc._set_process_label(self._calc_class.__name__)
```

Comment:
Try and deduplicate with code in Process._setup_db_record, particularly factor out input link creation and leave the rest as common